depth_image_proc/src/point_cloud_xyz.cpp

```diff
@@ -29,50 +29,13 @@
 // LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
 // ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
-#include <rclcpp/rclcpp.hpp>
-#include <image_transport/image_transport.hpp>
-#include <sensor_msgs/image_encodings.hpp>
-#include <image_geometry/pinhole_camera_model.h>
-#include <depth_image_proc/depth_conversions.hpp>
-#include <depth_image_proc/visibility.h>
```

Comment:
I am never in favor of this.  Every file should include what it uses; otherwise, you are just implicitly relying on headers files from elsewhere (which may change).  The same goes several times below.

Reply:
i dont really follow this logic. would you mind linking to some more in-depth info on this? I thought the point of header files was to define the object and its methods. without the header file, the cpp file is useless if you use this structure (which I thought was a cpp standard...?).

it sounds like what you're saying is to include the dependencies in the cpp file as well AND include the header file that defines the object. right? i guess ive been doing this wrong then.....

Reply:
> it sounds like what you're saying is to include the dependencies in the cpp file as well AND include the header file that defines the object. right?

Correct.

Reply:
This is the Include-What-You-Use principal. I tend to agree with this as it's part of the Google Style Guide, which the ROS2 Style Guide is based on. However, you should only include those files which you use in the appropriate file. For example, if you reference a message type in a function signature, it should be included in both the header and the source file (because it's used in both places). However, if you only use `std::sort` in the source file, only include `<algorithm>` in the source file.